src/components/Auth/SignupForm.tsx

```diff
@@ -1,33 +1,48 @@
 import PersonalInfoInput from 'components/common/PersonalInfoInput';
-import React, { useState } from 'react';
+import React, { useEffect } from 'react';
 import styled from 'styled-components';
+import { theme } from 'styles/theme';
+import { UserSignupRequest } from 'types/user';
 
-function SignupForm() {
-  const [signupInfo, setSignupInfo] = useState({
-    name: '',
-    nickname: '',
-    email: '',
-    emailConfirm: '',
-    password: '',
-    passwordConfirm: '',
-  });
+interface SignupFormProps {
+  signupInfo: UserSignupRequest;
+  handleOnChange: (type: keyof UserSignupRequest, value: string) => void;
+  handleComplete: (type: keyof UserSignupRequest, value: boolean) => void;
+}
+
+function SignupForm(props: SignupFormProps) {
+  const { signupInfo, handleOnChange, handleComplete } = props;
+  const checkPassword = () => signupInfo.password.value !== signupInfo.passwordConfirm.value;
```

Comment:
```suggestion
  const isPasswordEqual = () => signupInfo.password.value === signupInfo.passwordConfirm.value;
```

패스워드와 그 확인이 일치하는지 여부를 반환하는 함수이니까 이런 네이밍으로 하는건 어떨까요?
이렇게 했을 때 밑의 useEffect의 콜백에서도 무슨 일을 하는 지 좀 더 쉽게 판단할 수 있을 것 같아요!